File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        res = ""

        minn = 201

        for i in strs:

            if len(i)<minn:

                minn = len(i)

        k = 0

        for i in range(minn):

            che = strs[0][k]

            m = 1

            for j in range(1,len(strs)):

                if strs[j][k] == che:

                    m += 1

                else:

                    break

            k += 1

            if m == len(strs):

                res += che

            else:

                break

            

        return res
                       



        

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        string s;
        int minn = INT_MAX;
        for (string i:strs){
            minn = min(minn,int(i.size()));
        }
        if (strs.empty()) return "";
        for (int i=0;i<minn;i++){
            char a = strs[0][i];
            for (string c:strs){
                if (c[i] != a){
                    return s;
                }
            }
            s.push_back(a);
        }
        return s;

    }
};